test/app/views/integration_examples/member_var_in_controller.html.erb

```diff
@@ -0,0 +1 @@
+<%= render HelpersProxyComponent.new %>
```

Comment:
I'm curious if we could reproduce the test exception we saw above.

My hunch is that changing this to render the `ContainerComponent` would cause the same exception since `view_context` will sometimes be component instances instead of the original `view_context` that has helpers included, instance variables defined, etc.

Reply:
I'll try adding a new app/views/integration_examples view that renders `ContainerComponent` and corresponding controller endpoint to test this!

Reply:
Yep, it definitely fails! Pushed up the test in https://github.com/github/view_component/pull/926/commits/02428f3573bad20d5fc366e7ab16140136f4e188.

```
Error:
IntegrationTest#test_renders_container_component_that_uses_helpers:
ActionView::Template::Error: undefined method `message' for #<ContainerComponent:0x00007f984a18e680>
    /Users/cheshire137/code/github/view_component/test/app/components/helpers_proxy_component.html.erb:1:in `call'
    /Users/cheshire137/code/github/view_component/lib/view_component/compiler.rb:73:in `render_template_for'
    /Users/cheshire137/code/github/view_component/lib/view_component/base.rb:94:in `render_in'
```

Reply:
> view_context will sometimes be component instances instead of the original view_context

So we need a way to get at that original view context. 🤔 It feels strange to me that this `view_context` method would sometimes have a component instance in it, why does that happen?

Reply:
`ViewComponent::Base` inherits from `ActionView::Base` which implements `render` like: https://github.com/rails/rails/blob/91f0db1353e1b24e2d26ebbbadce510735f66848/actionview/lib/action_view/helpers/rendering_helper.rb#L43

When we call `render` in a component, it's passing `self` as an argument and not the original `view_context`. We could try to thread the original `view_context` all the way down, but I think that might introduce some new challenges (that are probably worth fixing anyways).

Reply:
Adding this line in ViewComponent::Base#helpers was informative:

```ruby
puts "has helpers? #{view_context.class <= ActionController::Base::HelperMethods ? 'yes' : 'no'}"
```

When I `bundle exec rake`, I see a lot of 'yes' lines and one 'no' right before the test error:

```
has helpers? yes
.....................has helpers? yes
has helpers? yes
.has helpers? no
E

Error:
IntegrationTest#test_renders_container_component_that_uses_helpers:
ActionView::Template::Error: undefined method `message' for #<ContainerComponent:0x00007fe712a673d0>
```